src/kubectlUtils.ts

```diff
@@ -447,6 +447,8 @@ export async function waitForRunningPod(kubectl: Kubectl, podName: string): Prom
         kubeChannel.showOutput(`pod/${podName} status: ${status}`);
         if (status === "Running") {
             return;
+        } else if (status.startsWith("Init")) {

```

Comment:
I apologize but I think I am missing your point on this one.  Probably looking at it wrong but I seeing "Init..." as the initContainer "running" to get the desired status update...guessing I am incorrect?  (I think I tried it w/o that test but I will try again.)

Reply:
My only concern (maybe it doesn't create any issue at all or it's me who's missing some point) is that the method `waitForRunningPod` should only return if the pod is in running state. This method is called from the debug action. 
So if i try to debug a pod and it gets returned because the init container is running is not the result i would expect. I want the actual container to be running, not the init one.
So my question is. I see this piece of code
```
        if (status === "Running") {
            return;
        } else if (status.startsWith("Init")) {
            return;
        } else if (!isTransientPodState(status)) {
            const logsResult = await kubectl.invokeCommand(`logs pod/${podName}`);
            kubeChannel.showOutput(`Failed to start the pod "${podName}". Its status is "${status}".
                Pod logs:\n${shellMessage(logsResult, "Unable to retrieve logs")}`);
            throw new Error(`Failed to start the pod "${podName}". Its status is "${status}".`);
        }
```
Wouldn't it be better to add your new part within the `isTransientPodState` function?
```
function isTransientPodState(status: string): boolean {
    return status === "ContainerCreating" || status === "Pending" || status === "Succeeded" || status.startsWith("Init") ;
}
```

and leave the above part as before
```
        if (status === "Running") {
            return;
        } else if (!isTransientPodState(status)) {
            const logsResult = await kubectl.invokeCommand(`logs pod/${podName}`);
            kubeChannel.showOutput(`Failed to start the pod "${podName}". Its status is "${status}".
                Pod logs:\n${shellMessage(logsResult, "Unable to retrieve logs")}`);
            throw new Error(`Failed to start the pod "${podName}". Its status is "${status}".`);
        }

```

Reply:
I think I have misunderstood the purpose of waiting on the pod/initContainer here.  I was assuming we might want functionality (logging, debugging) while the initContainer is running (I have worked on some projects with relatively long-lived initContainers at least long enough to warrant real-time logging/debugging).  I made the modification as suggested but I am not seeing the expected behavior.  I was expecting blocking until "Running" but I saw no difference.  Do you have any additional thoughts?

Reply:
That appears to be debugging related.  I have not been able to get (container) debugging working in the past with this plugin and noticed at least one TODO: in the code that calls that function.  I will make the changes you suggested for now but I suspect that once this is working (assuming it's not as in my own experience) that it will ultimately mean initContainers cannot be debugged if I am understanding the intent.